### PR TITLE
static webpack plugin

### PR DESCRIPTION
**Pull request by kylemac**
create + implement static webpack plugin.
- currently it supports all these [binary formats](https://github.com/sindresorhus/binary-extensions/blob/master/binary-extensions.json) + [image-extensions](https://github.com/arthurvr/image-extensions/blob/master/image-extensions.json)
- relies on our [forked file-loader](https://github.com/carrot/file-loader)

### CODE REVIEW

**Comment by jenius** [comment]
Any idea what happened on travis? Weird...

File: lib/plugins/static_plugin.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    this.files = util.getFilesFromGlob(compiler, this.opts)
+
+    // inject static files into webpack's pipeline
+    compiler.plugin('make', (compilation, done) => {
+      util.addFilesAsWebpackEntries(this.files, this.opts, compiler, compilation)
+        .then((res) => done())
+    })
+
+    // remove static assets from webpack pipeline, unless jadeTemplates option
+    // is present
```
Comment (jenius) [inline lib/plugins/static_plugin.js]: swap up this comment?

File: package.json [hunk 2]
```diff
@@ -15,13 +15,15 @@
     "babel-loader": "^6.2.0",
     "babel-preset-es2015": "^6.3.13",
     "babel-preset-stage-2": "^6.3.13",
+    "binary-extensions": "^1.4.0",
     "browser-sync": "^2.11.1",
     "browser-sync-webpack-plugin": "^1.0.1",
     "chalk": "^1.1.1",
     "css-loader": "^0.23.1",
+    "file-loader": "git@github.com:carrot/file-loader.git",
```
Comment (jenius) [inline package.json]: you can just use `carrot/file-loader` now and it will work :grin:

**Comment by kylemac** [comment]
yeah, one sec.

File: package.json (hunk 2 above)
Comment (kylemac) [inline package.json]: yup, thats what imma doin

**Comment by jenius** [comment]
:+1: lgtm

**Comment by kylemac** [comment]
have 1 more feature to add

**Pull request merged**